### PR TITLE
f/add-cf-turnstile-captcha

### PR DESCRIPTION
**Pull request by elias-summermatter**

### CODE REVIEW

File: libs/api/src/lib/challenges/cfTurnstile.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  ChallengeValidationProps,
+  ChallengeValidationReturn
+} from './challengeProvider'
+
+export class CfTurnstile implements ChallengeProvider {
+  constructor(private turnstile_secret_key: string) {}
+
+  async generateChallenge(testing = false): Promise<Challenge> {
+    return {
+      challengeID: 'not supported is turnstile!',
```
Comment (Itrulia) [inline libs/api/src/lib/challenges/cfTurnstile.ts:13]: Should probably not be set.

**Review by Itrulia — CHANGES_REQUESTED** [review]
Could we add a type or something to the challenge so the frontend knows what to do with it? I understand that in the future the algebraic one will be gone as it's essentially useless, but we could have different ones.

File: libs/api/src/lib/challenges/cfTurnstile.ts (hunk 1 above)
Comment (elias-summermatter) [reply libs/api/src/lib/challenges/cfTurnstile.ts:13]: @Itrulia  What do you mean by "should not be set"?

**Comment by elias-summermatter** [comment]
@Itrulia  Where would you return the type? Because for turnstle you don't need to do any request to the api you can just get the solution from  cf....

File: libs/api/src/lib/challenges/cfTurnstile.ts (hunk 1 above)
Comment (Itrulia) [reply libs/api/src/lib/challenges/cfTurnstile.ts]: The challengeID should not be set, if it's not supported.

Comment (elias-summermatter) [reply libs/api/src/lib/challenges/cfTurnstile.ts]: Is this possible on graphql level if it is required for algebraic but not for turnstile captcha?

Comment (Itrulia) [reply libs/api/src/lib/challenges/cfTurnstile.ts]: Yes, by using Unions (like Blocks). Which would then also require the type like I wrote in another comment. The frontend needs the type to know how to display it.

**Comment by elias-summermatter** [comment]
@Itrulia added a type could you maybe code the fe part that we can finish that off and make the media happy :-)

**Comment by Itrulia** [comment]
roger, will work on it today or tomrrow!

**Pull request closed without merging**